source/features/edit-comments-faster.tsx

```diff
@@ -8,7 +8,9 @@ import features from '.';
 
 function init(): void {
 	// Find editable comments first, then traverse to the correct position
-	observe('.js-comment.unminimized-comment .js-comment-update:not(.rgh-edit-comment)', {
+	observe((document.querySelector('.current-user span[title^="You are a"]') !== null || pageDetect.hasComments() ?
+		'.js-comment.unminimized-comment .js-comment-update:not(.rgh-edit-comment)' :
+		'.current-user.js-comment.unminimized-comment .js-comment-update:not(.rgh-edit-comment)'), {
```

Comment:
I would rather a if .... return

The above code is very hard to read

Reply:
Do you mean to change it like this?

```js
	let selector = '.current-user.js-comment.unminimized-comment .js-comment-update:not(.rgh-edit-comment)';
	if (document.querySelector('.current-user span[title^="You are a"]') !== null || pageDetect.hasComments()) {
		selector = '.js-comment.unminimized-comment .js-comment-update:not(.rgh-edit-comment)';
	}

	observe(selector, {
```

I'm asking, because I can't see where to use `return`, 
and if I use `if ... else`, then the `unicorn/prefer-ternary` lint rule is triggered.

Reply:
But, the change has to be done in the selector in the `observe(` line:

https://github.com/sindresorhus/refined-github/blob/575f41a7faacd5901065cfae634ce0399658d1cb/source/features/edit-comments-faster.tsx#L11-L12

which is before the `add()`.

How can I use the `selector` after the `add()` ?

Reply:
comment will be an element query based off the element

Reply:
We're using the selector itself to exclude comments that can't be edited, but it's failing because GitHub is including dead DOM here, so we have to improve the detection.

There are 2 types of comments that can be editable:

- every comment, if you're a moderator
- only your comment otherwise

I think we'll need something like:


```js
const preSelector = canEditDiscussion() ? '' : '.current-user';

observer(preSelector + '.js-comment.unminimized-comment .js-comment-update:not(.rgh-edit-comment)', {
```

However I'm not seeing #4400 at all

<img width="262" alt="Screen Shot 6" src="https://user-images.githubusercontent.com/1402241/119717057-0a299f00-be90-11eb-95ac-b2c4692391cf.png">

Probably because @cheap-glitch is a collaborator there but can't edit the comments. `canEditDiscussion` needs to be created with this in mind.

Reply:
> However I'm not seeing #4400 at all

I see it 😕 on every post there is an edit button

Reply:
@fregante, @yakov116 
Is it working ok for you with issue #4400 ?

This PR in its current state only applies to Discussion pages,
so you shouldn't have problem.